apps/explorer/src/app/components/txs/txs-per-block.tsx

```diff
@@ -1,37 +1,32 @@
-import type { ChainExplorerTxResponse } from '../../routes/types/chain-explorer-response';
 import { Routes } from '../../routes/route-names';
 import { DATA_SOURCES } from '../../config';
 import { RenderFetched } from '../render-fetched';
 import { TruncatedLink } from '../truncate/truncated-link';
 import { TxOrderType } from './tx-order-type';
 import { Table, TableRow, TableCell } from '../table';
 import { t, useFetch } from '@vegaprotocol/react-helpers';
+import type { BlockExplorerTransactions } from '../../routes/types/block-explorer-response';
 
 interface TxsPerBlockProps {
-  blockHeight: string | undefined;
+  blockHeight: string;
+  txCount: number;
 }
 
 const truncateLength = 14;
 
-export const TxsPerBlock = ({ blockHeight }: TxsPerBlockProps) => {
+export const TxsPerBlock = ({ blockHeight, txCount }: TxsPerBlockProps) => {
+  // TODO after https://github.com/vegaprotocol/vega/pull/6958/files is merged and deployed, use filter
+  // by block height instead
   const {
-    state: { data: decodedBlockData, loading, error },
-  } = useFetch<ChainExplorerTxResponse[]>(DATA_SOURCES.chainExplorerUrl, {
-    method: 'POST',
-    mode: 'cors',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify({
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      block_height: parseInt(blockHeight!),
-      node_url: `${DATA_SOURCES.tendermintUrl}/`,
-    }),
-  });
-
+    state: { data, loading, error },
+  } = useFetch<BlockExplorerTransactions>(
+    `${
+      DATA_SOURCES.blockExplorerUrl
+    }/transactions/before=${blockHeight.toString()}.0&limit=${txCount}`
```

Comment:
question (non-blocking): out of curiosity, what's that `.0` at the end of the before param's value?

Reply:
Proper answer: What's being passed *looks* like a block height, but it isn't - it's a cursor. `.0` is the index of the first item in the block.

Better answer: It won't work without it

Reply:
makes more sense, thank you!